### PR TITLE
Support for unknown variables

### PR DESCRIPTION
**Pull request by IvoTops**
Implemented support for getting no value for a variable and treating it as null. Evaluating x + y will be null if either x or y is unknown and not throw an error anymore.

- IFunction has new property CanHandleNullArguments. Set to true if the function needs and/or recognizes IsNull values. Set to false to let the code set the result automatically to IsNull if any argument for the function is null.
- Added IfNull() and NullIf() functions to handle null values inside your formulas! =>  IfNull(x,-1)<0 will be TRUE if x is unknown at evaluation time.
- NULL as text in a formula will be interpreted as a variable set to null, so =>  IfNull(null,1)  will evaluate to 1
- Results from Calcxxx are now changed to reflect nullable outcomes => double?, bool?,..
- New EvaluatorValueTypeKey Null.  Can be checked with IsNull property
- default(EvaluatorValue) and default(InternalEvaluatorValue) will be treated as a null  value
- implicit conversions with double?, datetime?, .. added
- Added testcases for Null handling
- Added VariablesUsed() function with some overloads to get a list of the variables a formula contains
- Fixed equals comparison for non number values
- Added Benchmark for evaluating solely evaluating performance (common usecase=> parse once, evaluate again and again with new values)
 
Documentation is not yet updated

### CODE REVIEW

File: NoStringEvaluating/Contract/INoStringEvaluator.cs [hunk 1]
```diff
@@ -5,416 +5,417 @@
 using NoStringEvaluating.Models.Values;
 using NoStringEvaluating.Nodes.Common;
 
-namespace NoStringEvaluating.Contract;
-
-/// <summary>
-/// Expression evaluator
-/// </summary>
-public interface INoStringEvaluator
+namespace NoStringEvaluating.Contract
```
Comment (KovtunV) [inline NoStringEvaluating/Contract/INoStringEvaluator.cs:8]: Don't change design for namespaces, please, fix in all files. Have you rebased successfully?

**Review by KovtunV — CHANGES_REQUESTED** [review]
Can't review other files until you fix namespace design

**Comment by IvoTops** [comment]
Ok, will fix. Is there an easy way to get the same settings in my visual
studio?

**Comment by KovtunV** [comment]
> Ok, will fix. Is there an easy way to get the same settings in my visual studio?

To change namespace design? Yes, it should be in Visual Studio

**Comment by IvoTops** [comment]
To rebase I fetched upstream from the master and that has the 'old' namespace design.  If I change the namespace design in visual studio and apply that it will also apply all my other code formatting settings. I see no option in vs to selectively apply only  the namespace design change. Could you run your coding style settings over the pull request and re-commit?

**Comment by KovtunV** [comment]
> To rebase I fetched upstream from the master and that has the 'old' namespace design. If I change the namespace design in visual studio and apply that it will also apply all my other code formatting settings. I see no option in vs to selectively apply only the namespace design change. Could you run your coding style settings over the pull request and re-commit?

is it possible? Think will make MR to your master branch with these fixes

**Comment by IvoTops** [comment]
If you think that works better please do that. I have been on GitHub for a
while but have only recently started contributing and using PR and such.

**Pull request merged**